packages/frontend/editor-ui/src/composables/commandBar/useWorkflowNavigationCommands.ts

```diff
@@ -0,0 +1,209 @@
+import { computed, ref, type Ref } from 'vue';
+import { useRouter, useRoute } from 'vue-router';
+import { N8nIcon } from '@n8n/design-system';
+import { useI18n } from '@n8n/i18n';
+import { useNodeTypesStore } from '@/stores/nodeTypes.store';
+import { useCredentialsStore } from '@/stores/credentials.store';
+import { useActionsGenerator } from '@/components/Node/NodeCreator/composables/useActionsGeneration';
+import debounce from 'lodash/debounce';
+import { VIEWS } from '@/constants';
+import type { IWorkflowDb } from '@/Interface';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import { useProjectsStore } from '@/stores/projects.store';
+import type { CommandGroup, CommandBarItem } from './types';
+
+const ITEM_ID = {
+	CREATE_WORKFLOW: 'create-workflow',
+	OPEN_WORKFLOW: 'open-workflow',
+};
+
+export function useWorkflowNavigationCommands(options: {
+	lastQuery: Ref<string>;
+	activeNodeId: Ref<string | null>;
+	currentProjectName: Ref<string>;
+}): CommandGroup {
+	const i18n = useI18n();
+	const { lastQuery, activeNodeId, currentProjectName } = options;
+	const nodeTypesStore = useNodeTypesStore();
+	const credentialsStore = useCredentialsStore();
+	const workflowsStore = useWorkflowsStore();
+	const projectsStore = useProjectsStore();
+
+	const router = useRouter();
+	const route = useRoute();
+
+	const { generateMergedNodesAndActions } = useActionsGenerator();
+
+	const workflowResults = ref<IWorkflowDb[]>([]);
+
+	const personalProjectId = computed(() => {
+		return projectsStore.myProjects.find((p) => p.type === 'personal')?.id;
+	});
+
+	function orderResultByCurrentProjectFirst<T extends IWorkflowDb>(results: T[]) {
+		const currentProjectId =
+			typeof route.params.projectId === 'string' ? route.params.projectId : personalProjectId.value;
+		return results.sort((a, b) => {
+			if (a.homeProject?.id === currentProjectId) return -1;
+			if (b.homeProject?.id === currentProjectId) return 1;
+			return 0;
+		});
+	}
+
+	const fetchWorkflows = debounce(async (query: string) => {
+		try {
+			const trimmed = (query || '').trim();
+			const nameSearchPromise = workflowsStore.searchWorkflows({
+				name: trimmed,
+			});
+
+			// Find matching node types from available nodes
+			const httpOnlyCredentials = credentialsStore.httpOnlyCredentialTypes;
+			const visibleNodeTypes = nodeTypesStore.allNodeTypes;
+			const { mergedNodes } = generateMergedNodesAndActions(visibleNodeTypes, httpOnlyCredentials);
+			const trimmedLower = trimmed.toLowerCase();
+			const matchedNodeTypeNames = Array.from(
+				new Set(
+					mergedNodes
+						.filter(
+							(node) =>
+								node.displayName?.toLowerCase().includes(trimmedLower) ||
+								node.name?.toLowerCase().includes(trimmedLower),
```

Comment:
This has a bug currently, when trimmedLower is an empty string this matches all nodes and ends up doing the search with all nodeTypes as matchedNodeTypeNames, we just don't pass it to the search.

I had this fixed on my local branch that I threw away after conflicts 😢 we can fix this in a followup pr